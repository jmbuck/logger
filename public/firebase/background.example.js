// Initialize Firebase
var config = {
    apiKey: "EXAMPLE",
    authDomain: "EXAMPLE",
    databaseURL: "EXAMPLE",
    projectId: "EXAMPLE",
    storageBucket: "EXAMPLE",
    messagingSenderId: "EXAMPLE"
};
firebase.initializeApp(config);

/**
 * initApp handles setting up the Firebase context and registering
 * callbacks for the auth status.
 *
 * The core initialization is in firebase.App - this is the glue class
 * which stores configuration. We provide an app name here to allow
 * distinguishing multiple app instances.
 *
 * This method also registers a listener with firebase.auth().onAuthStateChanged.
 * This listener is called when the user is signed in or out, and that
 * is where we update the UI.
 *
 * When signed in, we also authenticate to the Firebase Realtime Database.
 */
function initApp() {
    // Listen for auth state changes.
    firebase.auth().onAuthStateChanged(function(user) {
        console.log('User state change detected from the Background script of the Chrome Extension:', user);
    });
}

window.onload = function() {
    initApp();
};