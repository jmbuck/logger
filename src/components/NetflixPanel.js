import React, {Component} from 'react'
import '../css/common.css'
import exitIcon from "../img/x.svg"
import DataNav from './DataNav'
import {auth, retrieveFirebaseNetflixData} from "../rebase";

class NetflixPanel extends Component {

    constructor(props) {
        super(props);

        this.state = {
            timeTV : "...",
            timeMovies : "...",
            visitsTV : "...",
            visitsMovies : "..."
        }
    }

    retrieve = (user) => {
        if(user) {
            retrieveFirebaseNetflixData(user.uid, (data) => {
                this.setState(data)
            })
        }
    };

    componentWillMount() {
        if(auth.currentUser)
            this.retrieve(auth.currentUser);

        auth.onAuthStateChanged((user) => {
            this.retrieve(user);
        })
    }

    render() {
        return (
	        <div className="panel">
                <div className="panel-container">
                    <div className="panel-netflix">
						<DataNav {...this.props}/>
                        <div className="panel-center-content">
	                        <h1>Netflix Page</h1>
	                        <h3>Time Spent Watching TV -</h3>
	                        <div>{this.state.timeTV}</div>
	                        <h3>Time Spent Watching Movies -</h3>
	                        <div>{this.state.timeMovies}</div>
	                        <h3>Number of TV Shows Watched -</h3>
	                        <div>{this.state.visitsTV}</div>
	                        <h3>Number of Movies Watched -</h3>
	                        <div>{this.state.visitsMovies}</div>
                        </div>
                    </div>
                </div>
	        </div>
        )
    }
}

export default NetflixPanel