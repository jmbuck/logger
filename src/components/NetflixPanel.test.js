import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NetflixPanel from "./NetflixPanel";
import firebase from "firebase";

Enzyme.configure({ adapter: new Adapter() });

describe("NetflixPanel", () => {
    beforeAll(() => {
        firebase.auth = jest.fn().mockReturnValue({
            onAuthStateChanged: true
        });
    });

    it("should render component", () => {
        const component = Enzyme.shallow(<NetflixPanel />);

        expect(component).toMatchSnapshot();
    });
});

