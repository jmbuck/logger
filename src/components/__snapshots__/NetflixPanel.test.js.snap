// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetflixPanel should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NetflixPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="panel-container"
>
        <div
                className="panel-netflix"
        >
                <DataNav />
                <div
                        className="panel-center-content"
                >
                        <h1>
                                Netflix Page
                        </h1>
                        <h3>
                                Time Spent Watching TV -
                        </h3>
                        <div>
                                ...
                        </div>
                        <h3>
                                Time Spent Watching Movies -
                        </h3>
                        <div>
                                ...
                        </div>
                        <h3>
                                Number of TV Shows Watched -
                        </h3>
                        <div>
                                ...
                        </div>
                        <h3>
                                Number of Movies Watched -
                        </h3>
                        <div>
                                ...
                        </div>
                </div>
        </div>
</div>,
      "className": "panel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel-netflix"
>
          <DataNav />
          <div
                    className="panel-center-content"
          >
                    <h1>
                              Netflix Page
                    </h1>
                    <h3>
                              Time Spent Watching TV -
                    </h3>
                    <div>
                              ...
                    </div>
                    <h3>
                              Time Spent Watching Movies -
                    </h3>
                    <div>
                              ...
                    </div>
                    <h3>
                              Number of TV Shows Watched -
                    </h3>
                    <div>
                              ...
                    </div>
                    <h3>
                              Number of Movies Watched -
                    </h3>
                    <div>
                              ...
                    </div>
          </div>
</div>,
        "className": "panel-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DataNav />,
            <div
              className="panel-center-content"
>
              <h1>
                            Netflix Page
              </h1>
              <h3>
                            Time Spent Watching TV -
              </h3>
              <div>
                            ...
              </div>
              <h3>
                            Time Spent Watching Movies -
              </h3>
              <div>
                            ...
              </div>
              <h3>
                            Number of TV Shows Watched -
              </h3>
              <div>
                            ...
              </div>
              <h3>
                            Number of Movies Watched -
              </h3>
              <div>
                            ...
              </div>
</div>,
          ],
          "className": "panel-netflix",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Netflix Page
</h1>,
                <h3>
                  Time Spent Watching TV -
</h3>,
                <div>
                  ...
</div>,
                <h3>
                  Time Spent Watching Movies -
</h3>,
                <div>
                  ...
</div>,
                <h3>
                  Number of TV Shows Watched -
</h3>,
                <div>
                  ...
</div>,
                <h3>
                  Number of Movies Watched -
</h3>,
                <div>
                  ...
</div>,
              ],
              "className": "panel-center-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Netflix Page",
                },
                "ref": null,
                "rendered": "Netflix Page",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Time Spent Watching TV -",
                },
                "ref": null,
                "rendered": "Time Spent Watching TV -",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "...",
                },
                "ref": null,
                "rendered": "...",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Time Spent Watching Movies -",
                },
                "ref": null,
                "rendered": "Time Spent Watching Movies -",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "...",
                },
                "ref": null,
                "rendered": "...",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Number of TV Shows Watched -",
                },
                "ref": null,
                "rendered": "Number of TV Shows Watched -",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "...",
                },
                "ref": null,
                "rendered": "...",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Number of Movies Watched -",
                },
                "ref": null,
                "rendered": "Number of Movies Watched -",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "...",
                },
                "ref": null,
                "rendered": "...",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel-container"
>
          <div
                    className="panel-netflix"
          >
                    <DataNav />
                    <div
                              className="panel-center-content"
                    >
                              <h1>
                                        Netflix Page
                              </h1>
                              <h3>
                                        Time Spent Watching TV -
                              </h3>
                              <div>
                                        ...
                              </div>
                              <h3>
                                        Time Spent Watching Movies -
                              </h3>
                              <div>
                                        ...
                              </div>
                              <h3>
                                        Number of TV Shows Watched -
                              </h3>
                              <div>
                                        ...
                              </div>
                              <h3>
                                        Number of Movies Watched -
                              </h3>
                              <div>
                                        ...
                              </div>
                    </div>
          </div>
</div>,
        "className": "panel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="panel-netflix"
>
            <DataNav />
            <div
                        className="panel-center-content"
            >
                        <h1>
                                    Netflix Page
                        </h1>
                        <h3>
                                    Time Spent Watching TV -
                        </h3>
                        <div>
                                    ...
                        </div>
                        <h3>
                                    Time Spent Watching Movies -
                        </h3>
                        <div>
                                    ...
                        </div>
                        <h3>
                                    Number of TV Shows Watched -
                        </h3>
                        <div>
                                    ...
                        </div>
                        <h3>
                                    Number of Movies Watched -
                        </h3>
                        <div>
                                    ...
                        </div>
            </div>
</div>,
          "className": "panel-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <DataNav />,
              <div
                className="panel-center-content"
>
                <h1>
                                Netflix Page
                </h1>
                <h3>
                                Time Spent Watching TV -
                </h3>
                <div>
                                ...
                </div>
                <h3>
                                Time Spent Watching Movies -
                </h3>
                <div>
                                ...
                </div>
                <h3>
                                Number of TV Shows Watched -
                </h3>
                <div>
                                ...
                </div>
                <h3>
                                Number of Movies Watched -
                </h3>
                <div>
                                ...
                </div>
</div>,
            ],
            "className": "panel-netflix",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Netflix Page
</h1>,
                  <h3>
                    Time Spent Watching TV -
</h3>,
                  <div>
                    ...
</div>,
                  <h3>
                    Time Spent Watching Movies -
</h3>,
                  <div>
                    ...
</div>,
                  <h3>
                    Number of TV Shows Watched -
</h3>,
                  <div>
                    ...
</div>,
                  <h3>
                    Number of Movies Watched -
</h3>,
                  <div>
                    ...
</div>,
                ],
                "className": "panel-center-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Netflix Page",
                  },
                  "ref": null,
                  "rendered": "Netflix Page",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Time Spent Watching TV -",
                  },
                  "ref": null,
                  "rendered": "Time Spent Watching TV -",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "...",
                  },
                  "ref": null,
                  "rendered": "...",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Time Spent Watching Movies -",
                  },
                  "ref": null,
                  "rendered": "Time Spent Watching Movies -",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "...",
                  },
                  "ref": null,
                  "rendered": "...",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Number of TV Shows Watched -",
                  },
                  "ref": null,
                  "rendered": "Number of TV Shows Watched -",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "...",
                  },
                  "ref": null,
                  "rendered": "...",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Number of Movies Watched -",
                  },
                  "ref": null,
                  "rendered": "Number of Movies Watched -",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "...",
                  },
                  "ref": null,
                  "rendered": "...",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
